test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1

```diff
@@ -1130,6 +1144,108 @@ Describe "Invoke-WebRequest tests" -Tags "Feature" {
             $result.Files[0].ContentType | Should Be 'text/plain'
             $result.Files[0].Content | Should Match 'TestContent'
         }
+
+        It "Verifies Invoke-WebRequest -Form supports string values" {
+            $form = @{TestString = "TestValue"}
+            $uri = Get-WebListenerUrl -Test 'Multipart'
+            $response = Invoke-WebRequest -Uri $uri -Form $form -Method 'POST'
+            $result = $response.Content | ConvertFrom-Json
+
+            $result.Headers.'Content-Type' | Should Match 'multipart/form-data'
+            $result.Items.TestString.Count | Should Be 1
+            $result.Items.TestString[0] | Should Be 'TestValue'
+        }
+
+        It "Verifies Invoke-WebRequest -Form supports a collection of string values" {
+            $form = @{TestStrings = "TestValue", "TestValue2"}
+            $uri = Get-WebListenerUrl -Test 'Multipart'
+            $response = Invoke-WebRequest -Uri $uri -Form $form -Method 'POST'
+            $result = $response.Content | ConvertFrom-Json
+
+            $result.Headers.'Content-Type' | Should Match 'multipart/form-data'
+            $result.Items.TestStrings.Count | Should Be 2
+            $result.Items.TestStrings[0] | Should Be 'TestValue'
+            $result.Items.TestStrings[1] | Should Be 'TestValue2'
+        }
+
+        It "Verifies Invoke-WebRequest -Form supports file values" {
+            $form = @{TestFile = [System.IO.FileInfo]$file1Path}
+            $uri = Get-WebListenerUrl -Test 'Multipart'
+            $response = Invoke-WebRequest -Uri $uri -Form $form -Method 'POST'
+            $result = $response.Content | ConvertFrom-Json
+
+            $result.Headers.'Content-Type' | Should Match 'multipart/form-data'
+            $result.Files.Count | Should Be 1
+
+            $result.Files[0].Name | Should Be "TestFile"
+            $result.Files[0].FileName | Should Be $file1Name
+            $result.Files[0].ContentType | Should Be 'application/octet-stream'
+            $result.Files[0].Content | Should Match $file1Contents
```

Comment:
If we want get the same content I think we should use `BeExactly`.

Reply:
We have to use match here because kestrel doesn't act the same on all platforms.

Reply:
Please clarify, if I send binary file (zip) I get it corrupted? Can we then trust these tests?

Reply:
It's not on the sending PS side. It has something to do with the result JSON serialization. Sometimes the newline is trimmed and sometimes not. It's a rendering issue. We can improve the trust of these tests with file hashes, return an array of bytes, or a base64.

Reply:
As far as I understand ' application/octet-stream ' is a binary stream - then we have a serious bug in Kestrel 😕

Reply:
You misunderstand. This is when the stream is converted to test and again to JSON. Kestrels gets it just fine, but when the string is converted to JSON the behavior is a bit unstable between platforms. like I said, we should use one of the three methods to improve trust in the test. its not trivial work though and it will affect other tests. if we want to improve the trust, we should have a separate issue to include one of the three options.

Reply:
> when the string is converted to JSON the behavior is a bit unstable between platforms.
For full my understanding  - where is the wrong code?

Reply:
It is somewhere between Kestrel and ASP.NET JSON result. no clue.. beyond my skill to track down. I did some basic testing where I had the server save the uploaded file to disk and manually ran hashes on the source and upload file to ensure they match. no problem there.. so something happens between the filestream being converted to text and then to a JSON result and then sent as a HttpResponse. I can see it in tcpdump that it's coming out of the server wrong.. so it's not a rendering issue on the PowerShell side either.  

That stack is too deep for my skill level to look into. I can't reliably reproduce it either. It's why the other existing multipart tests are doing match on the the rendered file content. But I'm not too concerned.. rendering binary to text is always a bit wonky... a better test from the start would be to have provide SHA256 hash for the byte stream. that should match the sha256 hash of the FileStream on the source side. 

I don't want to do it in this PR because the change is not trivial and will affect the other tests. I'd rather keep with the `match` here and address the trust improvements in another PR. (the stream is forward only, so if I want to provide text contents as well as hash I have to introduce a buffer, or I have to switch to just the hash.. either way has an impact on the existing tests.)

Someday when I have time to become an expert at debugging .NET Core... I might be able to actually find the root cause of such things. 😆

Reply:
Many thanks! I think we should fix the issue in ??? - your description is great for openning Issue (in CoreFX repo if Kestrel use CoreFX System.Json.
After that we can use safely `BeExactly` in this tests. I believe we can't complicate our tests if we get the fix.
Also we already have meta Issue for tracking web cmdlets - we should add this in the Issue.

Reply:
Perhaps we should add a comment here